kernel/userspace.c

```diff
@@ -585,8 +585,8 @@ static void dump_permission_error(struct z_object *ko)
 	int index = thread_index_get(_current);
 	LOG_ERR("thread %p (%d) does not have permission on %s %p",
 		_current, index,
-		otype_to_str(ko->type), ko->name);
-	LOG_HEXDUMP_ERR(ko->perms, sizeof(ko->perms), "permission bitmap");
+		otype_to_str((enum k_objects)ko->type), ko->name);
+	LOG_HEXDUMP_ERR(&ko->perms, sizeof(ko->perms), "permission bitmap");
```

Comment:
Why are we adding a `&` here?

Reply:
It sounds more correct to me as the helper receive the pointer to an area and its size, but if this is not liked I can revert it.